test-interface/src/main/scala/scala/scalanative/testinterface/TestMainBase.scala

```diff
@@ -148,3 +194,29 @@ abstract class TestMainBase {
     override def trace(t: Throwable): Unit = log(Log.Level.Trace, "", Some(t))
   }
 }
+
+object TestMainBase {
+  private val faultHandler = CFunctionPtr.fromFunction1(handleSegFault _)
+  private def handleSegFault(id: Int): Unit = {
+    //making ids stable i.e. vals instead fo defs
+    //for use in the match
+    val SIGSEGV = signal.SIGSEGV
+    val SIGFPE  = signal.SIGFPE
+
+    val retCode = id match {
+      case `SIGSEGV` =>
+        Console.err.println("Segmentation fault")
+        139
+      case `SIGFPE` =>
+        Console.err.println("Erroneous Arithmetic Operation")
+        136
+      case _ =>
+        Console.err.println("Unknown fault")
+        -1
```

Comment:
Shouldn't it return `128 + id`? Otherwise, sbt will just say it exited with `-1`, which is wrong.

Reply:
Same here.